trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/get-link-to-premium-payments-tab.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Generates a link to the Premium Payments tab for a given utilisation report,
+ * including selected fee record IDs as query parameters if within URL length limit.
+ * @param reportId - The ID of the utilisation report.
+ * @param selectedFeeRecordIds - An array of selected fee record IDs to include in the URL.
+ * @returns A string representing the URL for the Premium Payments tab.
+ */
+export const getLinkToPremiumPaymentsTab = (reportId: string, selectedFeeRecordIds: number[]): string => {
+  const urlWithoutParams = `/utilisation-reports/${reportId}`;
+
+  if (selectedFeeRecordIds.length === 0) {
+    return urlWithoutParams;
+  }
+
+  const commaSeparatedFeeRecordIds = selectedFeeRecordIds.join(',');
+
+  const urlParams = new URLSearchParams({
+    selectedFeeRecordIds: commaSeparatedFeeRecordIds,
+  }).toString();
+
+  const urlWithParams = `${urlWithoutParams}?${urlParams}`;
+
+  // "Recommended Security and Performance Max: 2048 CHARACTERS" - https://stackoverflow.com/a/48230425
+  const maxUrlCharacters = 2048;
+  const baseUrlCharacterAllowance = 150;
+  const urlWithParamsMaxLength = maxUrlCharacters - baseUrlCharacterAllowance;
```

Comment:
Can we move these and the releated documentation to constants?

So this can become simply 

```ts
if (urlWithParams.length > URL_PARAMS_MAX_LENGTH) {
  // console.error()

  return urlWithoutParams;
}
```

Would make for easier consumption if we need to check URL params in other areas

Reply:
Moved these back to constants and added docs to each thanks